lib/AppInfo/Application.php

```diff
@@ -185,6 +192,20 @@ public function register(IRegistrationContext $context): void {
 		});
 
 		$context->registerServiceAlias(IUserMappingManager::class, UserMappingManager::class);
+
+		$context->registerService('AuthorizedAdminSettingMiddleware', function ($c) {
+			return new AuthorizedAdminSettingMiddleware(
+				$c->query(IControllerMethodReflector::class),
+				$c->query(DelegationService::class),
+				$c->query(IRequest::class),
+				$c->query(LoggerInterface::class),
+				$c->query(ControllerMethodReflector::class),
+				$c->query(AuthorizedGroupMapper::class),
+				$c->query(IUserSession::class),
+				$c->query(IGroupManager::class),
+			);
+		});
```

Comment:
```suggestion
```

Reply:
also not needed

Reply:
Yes, I need these lines to inject these methods into the AuthorizedAdminSettingMiddleware class

Reply:
Or, I have to replace these lines with this one : `$context->registerMiddleware(AuthorizedAdminSettingMiddleware::class);` ?

Is it compatible from NC23 ?

Reply:
Just using `$context->registerMiddleware(AuthorizedAdminSettingMiddleware::class)` should work, no need to register a service

Reply:
It's fixed !